internal/archive/archive.go

```diff
@@ -143,6 +143,49 @@ func ReadTarEntry(tarPath string, entryPath ...string) (*tar.Header, []byte, err
 	return nil, nil, fmt.Errorf("could not find entry path '%s' in tar", entryPath)
 }
 
+func ListTarContents(tarPath string) ([]tar.Header, error) {
```

Comment:
It seems like this function is only used by tests. We should probably move it to `testhelpers`.